lib/services/local/access_list_test.go

```diff
@@ -145,8 +145,176 @@ func TestAccessListDedupeOwnersBackwardsCompat(t *testing.T) {
 	})
 	require.NoError(t, err)
 
-	service, err := NewAccessListService(backend.NewSanitizer(mem), clock)
+	service := newAccessListService(t, mem, clock)
+
+	// Start with no license.
+	modules.SetTestModules(t, &modules.TestModules{
+		TestFeatures: modules.Features{
+			IdentityGovernance: false,
+		},
+	})
+
+	accessList1 := newAccessList(t, "accessList1", clock)
+	accessList2 := newAccessList(t, "accessList2", clock)
+	accessList3 := newAccessList(t, "accessList3", clock)
+
+	cmpOpts := []cmp.Option{
+		cmpopts.IgnoreFields(header.Metadata{}, "ID", "Revision"),
+	}
+
+	// First create is free.
+	accessList, err := service.UpsertAccessList(ctx, accessList1)
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(accessList1, accessList, cmpOpts...))
+
+	// Second create should return an error.
+	accessList, err = service.UpsertAccessList(ctx, accessList2)
+	require.True(t, trace.IsAccessDenied(err), "expected access denied / license limit error, got %v", err)
+	require.ErrorContains(t, err, "reached its limit")
+
+	// Double check only be one access list exists.
+	out, err := service.GetAccessLists(ctx)
+	require.NoError(t, err)
+	require.Len(t, out, 1)
+	require.Equal(t, out[0].Metadata.Name, "accessList1")
+
+	// Updating existing access list should be allowed.
+	accessList1.Spec.Description = "changing description"
+	accessList, err = service.UpsertAccessList(ctx, accessList1)
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(accessList1, accessList, cmpOpts...))
+
+	// Delete the one access list.
+	err = service.DeleteAccessList(ctx, "accessList1")
+	require.NoError(t, err)
+
+	// Create the same list again.
+	accessList, err = service.UpsertAccessList(ctx, accessList1)
 	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(accessList1, accessList, cmpOpts...))
+
+	// Enable license
+	modules.SetTestModules(t, &modules.TestModules{
+		TestFeatures: modules.Features{
+			IdentityGovernance: true,
+		},
+	})
+
+	// Should be able to create the rest of access lists.
+	accessList, err = service.UpsertAccessList(ctx, accessList2)
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(accessList2, accessList, cmpOpts...))
+
+	accessList, err = service.UpsertAccessList(ctx, accessList3)
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(accessList3, accessList, cmpOpts...))
+
+	// Fetch all access lists.
+	out, err = service.GetAccessLists(ctx)
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff([]*accesslist.AccessList{accessList1, accessList2, accessList3}, out, cmpOpts...))
+}
+
+func TestAccessListCreateLimit_UpsertAccessListWithMembers(t *testing.T) {
```

Comment:
This is a long test. Can you add a comment and explain the scenario that we're testing here?